### DISCUSSION TITLE
[20 Marzo 2023] - Modifica CPU

### DISCUSSION THREAD

**Discussion by rimaout** [CS-Swap/Architettura-degli-Elaboratori#31] [Modifica Architettura]
<img width="861" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/7039ef2a-cbda-4945-ba63-b3b203f000d0">
<img width="855" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/63fd7c57-546e-44c3-85ba-d2a0faf3e000">

<details><summary>Soluzione Prof</summary><p> 
<img width="855" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/6c185f55-cea0-4f1c-8009-5764330058ed">
</p></details>

>**[Altre Soluzioni](https://github.com/sapienzastudentsnetwork/architettura-degli-elaboratori/discussions/69)**

**Comment by rimaout** [comment]
### 1) Modifica CPU
1. Leggo $val1
2. Leggo dalla memoria val_address * 4 + 0x10010000 (val2)
3. Confronto val1 con val2:
    - se val1  >  val2: salvo in outreg val1
    - se val1 <= val2: salvo in outreg val2
  
**oss:** val1>val2 $\implies$ val2<val1 $\implies$ val2-val1<0, ovvero il risultato dell'operazione val2 - val1 deve essere negativo (ovvero il 31° bit deve essere uguale a 1)

<img width="812" alt="54A608F8-9CC1-45DA-A47F-A931DD7924C9" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/413c5277-4e13-46bf-b53f-6acabec0a257">

### 2)

<img width="1080" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/7b7e821e-96a5-4c32-ad64-7bc56cfb9cdb">

### 3)

<img width="800" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/eeec463b-20d8-4f89-95d6-ab6fe7631f0c">

**Comment by thegeek-sys** [comment]
![Screenshot 2024-06-03 alle 10 58 27](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/55345633/713ecb5d-f94f-4e86-872b-79c9d7768bf5)
![Screenshot 2024-06-03 alle 10 58 53](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/55345633/027e4def-938e-4799-bcda-7fb6fb4a36b5)
